test/e2e/util_gatewayapi.go

```diff
@@ -0,0 +1,647 @@
+			return false, nil
+		}
+		t.Logf("found subscription %s at installed version %s", subscription.Name, subscription.Status.InstalledCSV)
+		return true, nil
+	})
+	return err
+}
+
+// assertOSSMOperator checks if the OSSM Istio operator gets successfully installed
+// and returns an error if not.
+func assertOSSMOperator(t *testing.T) error {
+	t.Helper()
+	dep := &appsv1.Deployment{}
+	ns := types.NamespacedName{Namespace: openshiftOperatorsNamespace, Name: openshiftIstioOperatorDeploymentName}
+
+	// Get the Istio operator deployment.
+	err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), ns, dep); err != nil {
+			t.Logf("failed to get deployment %v, retrying...", ns)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("error finding deployment %v: %v", ns, err)
+	}
+
+	// Get the istio-operator pod.
+	podlist, err := getPods(t, kclient, dep)
+	if err != nil {
+		return fmt.Errorf("error finding pod for deployment %v: %v", ns, err)
+	}
+	if len(podlist.Items) > 1 {
+		return fmt.Errorf("too many pods for deployment %v: %d", ns, len(podlist.Items))
+	}
+	pod := podlist.Items[0]
+	if pod.Status.Phase != corev1.PodRunning {
+		return fmt.Errorf("OSSM operator failure: pod %s is not running, it is %v", pod.Name, pod.Status.Phase)
+	}
+
+	t.Logf("found OSSM operator pod %s/%s to be %s", pod.Namespace, pod.Name, pod.Status.Phase)
+	return nil
+}
+
+// assertIstiodProxy checks if the OSSM Istiod proxy gets successfully installed
+// and returns an error if not.
+func assertIstiodProxy(t *testing.T) error {
+	t.Helper()
+	dep := &appsv1.Deployment{}
+	ns := types.NamespacedName{Namespace: openshiftIngressNamespace, Name: openshiftIstiodDeploymentName}
+
+	// Get the Istiod proxy deployment.
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), ns, dep); err != nil {
+			t.Logf("failed to get deployment %v, retrying...", ns)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("error finding deployment %v: %v", ns, err)
+	}
+
+	// Get the istiod pod.
+	podlist, err := getPods(t, kclient, dep)
+	if err != nil {
+		return fmt.Errorf("error finding pod for deployment %v: %v", ns, err)
+	}
+	if len(podlist.Items) > 1 {
+		return fmt.Errorf("too many pods for deployment %v: %d", ns, len(podlist.Items))
+	}
+	pod := podlist.Items[0]
+	if pod.Status.Phase != corev1.PodRunning {
+		return fmt.Errorf("Istiod proxy failure: pod %s is not running, it is %v", pod.Name, pod.Status.Phase)
+	}
+
+	t.Logf("found istiod pod %s/%s to be %s", pod.Namespace, pod.Name, pod.Status.Phase)
+	return nil
+}
+
+// assertGatewayClassSuccessful checks if the gateway class was created and accepted successfully
+// and returns an error if not.
+func assertGatewayClassSuccessful(t *testing.T, name string) (error, *gwapi.GatewayClass) {
```

Comment:
I believe the go standard is to return errors last.

```suggestion
func assertGatewayClassSuccessful(t *testing.T, name string) (*gwapi.GatewayClass, error) {
```

Reply:
It's definitely better style.  Don't know what I was thinking.